tests/t0118_gateway_car_test.go

```diff
@@ -5,52 +5,595 @@ import (
+				Query("format", "car").
+				Query("dag-scope", "entity").
+				Query("entity-bytes", "0:*"),
+			Response: Expect().
+				Status(200).
+				Body(
+					IsCar().
+						HasRoot(subdirWithMixedBlockFiles.MustGetCid()).
+						HasBlocks(flattenStrings(t,
+							subdirWithMixedBlockFiles.MustGetCid(),
+							subdirWithMixedBlockFiles.MustGetCid("subdir"),
+							subdirWithMixedBlockFiles.MustGetCid("subdir", "multiblock.txt"),
+							subdirWithMixedBlockFiles.MustGetChildrenCids("subdir", "multiblock.txt"),
+						)...).
+						Exactly().
+						InThatOrder(),
+				),
+		},
+		{
+			Name: "GET CAR with entity-bytes of a UnixFS directory",
+			Hint: `
+				dag-scope=entity&entity-bytes=from:to should return a CAR file with all the blocks needed to enumerate contents of
+				a UnixFS directory at the end of the specified path if the terminal element is a directory
+				(i.e. entity-bytes is effectively optional if the entity is not a file)
+			`,
+			Request: Request().
+				Path("/ipfs/{{cid}}", singleLayerHamtMultiBlockFilesFixture.MustGetCidWithCodec(0x70)).
+				Query("format", "car").
+				Query("dag-scope", "entity").
+				Query("entity-bytes", "0:*"),
+			Response: Expect().
+				Status(200).
+				Body(
+					IsCar().
+						HasRoot(singleLayerHamtMultiBlockFilesFixture.MustGetCid()).
+						HasBlocks(
+							flattenStrings(t,
+								singleLayerHamtMultiBlockFilesFixture.MustGetCid(),
+								singleLayerHamtMultiBlockFilesFixture.MustGetCidsInHAMT())...,
+						).
+						Exactly().
+						InThatOrder(),
+				),
+		},
+		{
+			Name: "GET CAR with entity-bytes equivalent to a HTTP Range Request from the middle of a file to the end",
+			Hint: `
+				The response MUST contain only the minimal set of blocks necessary for fulfilling the range request
+			`,
+			Request: Request().
+				Path("/ipfs/{{cid}}", subdirWithMixedBlockFiles.MustGetCidWithCodec(0x70, "subdir", "multiblock.txt")).
+				Query("format", "car").
+				Query("dag-scope", "entity").
+				Query("entity-bytes", "512:*"),
+			Response: Expect().
+				Status(200).
+				Body(
+					IsCar().
+						HasRoot(subdirWithMixedBlockFiles.MustGetCid("subdir", "multiblock.txt")).
+						HasBlocks(
+							flattenStrings(t,
+								subdirWithMixedBlockFiles.MustGetCid("subdir", "multiblock.txt"),
+								subdirWithMixedBlockFiles.MustGetChildrenCids("subdir", "multiblock.txt")[2:])...,
+						).
+						Exactly().
+						InThatOrder(),
+				),
+		},
+		{
+			Name: "GET CAR with entity-bytes equivalent to a HTTP Range Request for the middle of a file",
+			Hint: `
+				The response MUST contain only the minimal set of blocks necessary for fulfilling the range request
+			`,
+			Request: Request().
+				Path("/ipfs/{{cid}}", subdirWithMixedBlockFiles.MustGetCidWithCodec(0x70, "subdir", "multiblock.txt")).
+				Query("format", "car").
+				Query("dag-scope", "entity").
+				Query("entity-bytes", "512:1023"),
```

Comment:
:thought_balloon: this works for all-in-one implementations like boxo/gateway, but won't be enough for Saturn.  I don't want to block on this – filled https://github.com/ipfs/gateway-conformance/issues/75 as a follow-up work.